com.unity.ml-agents/Runtime/Demonstrations/DemonstrationRecorder.cs

```diff
@@ -70,6 +75,11 @@ void Update()
             {
                 LazyInitialize();
             }
+
+            if (m_DemoWriter.NumSteps >= numStepsToRecord)
```

Comment:
Should we only quit if `NumStepsToRecord > 0` too?

Reply:
added >0 check. good idea